From 05a6a0a9961a08e9b025742dea5d922abdc061b8 Mon Sep 17 00:00:00 2001
From: Han Gao <gaohan@iscas.ac.cn>
Date: Sun, 1 Sep 2024 01:40:15 +0800
Subject: [PATCH 266/400] DPU: remove panel unprepare process in disable dsi
 process

Kernel 6.6 DRM framework does not require panel unprepare before disabling DSI.
correct disable process: panel disable -> encoder disable -> dsi disable -> panel unprepare.

Signed-off-by: Han Gao <gaohan@iscas.ac.cn>
---
 drivers/gpu/drm/verisilicon/dw_mipi_dsi.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/verisilicon/dw_mipi_dsi.c b/drivers/gpu/drm/verisilicon/dw_mipi_dsi.c
index 6c3ed4234c2a..f7fa83147793 100644
--- a/drivers/gpu/drm/verisilicon/dw_mipi_dsi.c
+++ b/drivers/gpu/drm/verisilicon/dw_mipi_dsi.c
@@ -833,7 +833,8 @@ static void bridge_post_disable(struct drm_bridge *bridge)
 	 * This needs to be fixed in the drm_bridge framework and the API
 	 * needs to be updated to manage our own call chains...
 	 */
-	primary->panel_bridge->funcs->post_disable(primary->panel_bridge);
+	if (primary->panel_bridge->funcs->post_disable)
+		primary->panel_bridge->funcs->post_disable(primary->panel_bridge);
 
 	if (primary->secondary_dsi)
 		dw_mipi_dsi_disable(primary->secondary_dsi);
-- 
2.43.0

